SHOWRAW.BAS

```diff
@@ -1,32 +1,116 @@
+SUB CyclePalette
+        DIM blue AS LONG
+        DIM green AS LONG
+        DIM red AS LONG
+        DIM pcolour AS LONG
+        DIM pvalues1(255) AS LONG
+        DIM pvalues2(255) AS LONG
+        DIM pvalues3(255) AS LONG
+        DIM pvalues4(255) AS LONG
+        FOR i = 0 TO 63
+                blue = i
+                green = i / 2
+                red = i / 3
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues1(i) = pcolour
+                pvalues1(i + 128) = pcolour
+                blue = i / 2
+                green = i / 3
+                red = i
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues2(i) = pcolour
+                pvalues2(i + 128) = pcolour
+                blue = i / 3
+                green = i
+                red = i / 2
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues3(i) = pcolour
+                pvalues3(i + 128) = pcolour
+                blue = i
+                green = i
+                red = i
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues4(i) = pcolour
+                pvalues4(i + 128) = pcolour
+        NEXT i
+        FOR i = 63 TO 0 STEP -1
+                blue = i
+                green = i / 2
+                red = i / 3
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues1(i + 64) = pcolour
+                pvalues1(i + 192) = pcolour
+                blue = i / 2
+                green = i / 3
+                red = i
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues2(i + 64) = pcolour
+                pvalues2(i + 192) = pcolour
+                blue = i / 3
+                green = i
+                red = i / 2
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues3(i + 64) = pcolour
+                pvalues3(i + 192) = pcolour
+                blue = i
+                green = i
+                red = i
+                pcolour = blue * 65536 + green * 256 + red
+                pvalues4(i + 64) = pcolour
+                pvalues4(i + 192) = pcolour
+       
+        NEXT i
 
-  OPEN filename FOR BINARY AS #1
-  DIM colour AS INTEGER
-  
-  FOR y = 0 TO h - 1
-    
-    FOR x = 0 TO w - 1
-    
-      position = (x + (Y * w))
-      
-      GET #1, position + 1, colour
-      
-      PSET (x, y), colour
+        index = 0
+        DO WHILE INKEY$ = ""
+                IF index = 0 THEN
+                        PALETTE USING pvalues4
+                ELSEIF index = 1 THEN
+                        PALETTE USING pvalues3
+                ELSEIF index = 2 THEN
+                        PALETTE USING pvalues2
+                ELSE
+                        PALETTE USING pvalues1
+                END IF
+
+                index = index + 1
+                FOR i = 0 TO 10000
+                NEXT i
```

Comment:
Is this just a spin to pause the display?

Reply:
Yeah sure!

Reply:
Done